akka-typed-tests/src/test/scala/akka/typed/cluster/sharding/ClusterShardingSpec.scala

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+
+package akka.typed.cluster.sharding
+
+import akka.cluster.sharding.ShardCoordinator.LeastShardAllocationStrategy
+import akka.typed.{ ActorRef, ActorSystem, Props, TypedSpec }
+import akka.typed.cluster.Cluster
+import akka.typed.internal.adapter.ActorSystemAdapter
+import akka.typed.scaladsl.Actor
+import akka.typed.scaladsl.adapter._
+import akka.typed.testkit.TestKitSettings
+import akka.typed.testkit.scaladsl.TestProbe
+import com.typesafe.config.ConfigFactory
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.duration._
+
+object ClusterShardingSpec {
+  val config = ConfigFactory.parseString(
+    """
+      akka.actor.provider = cluster
+      
+      // akka.loglevel = debug
+      
+      akka.remote.artery.enabled = true
+      akka.remote.netty.tcp.port = 0
+      akka.remote.artery.canonical.port = 0
```

Comment:
please add `akka.remote.artery.canonical.hostname = 127.0.0.1`, otherwise it will fail on my mac